campaign-launcher/interfaceV2/src/providers/Web3AuthProvider.tsx

```diff
@@ -0,0 +1,142 @@
+import {
+  createContext,
+  FC,
+  PropsWithChildren,
+  useContext,
+  useState,
+  useEffect,
+} from 'react';
+
+import { useAccount, useSignMessage } from 'wagmi';
+
+import { REFRESH_FAILURE_EVENT } from '../api/httpClient';
+import recordingApi from '../api/recordingApi';
+import { tokenManager, TokenData } from '../utils/TokenManager';
+
+type Nonce = 'signup' | string;
+
+type Web3AuthContextType = {
+  isAuthenticated: boolean;
+  isLoading: boolean;
+  signIn: () => Promise<void>;
+  logout: () => Promise<void>;
+};
+
+const Web3AuthContext = createContext<Web3AuthContextType>(
+  {} as Web3AuthContextType
+);
+
+const getNonce = async (address: `0x${string}` | undefined) => {
+  const response = await recordingApi.post<Nonce>(`/auth/nonce`, { address });
```

Comment:
Instead of using `post(`/auth/nonce`)` it's better to have `recordingApi.getNonce`, so you won't need to declare `getNonce` here, just in the API wrapper with convenient types already + only one place knows about all API routes and schema